### PR TITLE
Upgrade to rails 6

### PR DESCRIPTION
**Pull request by cdccollins**
Upgrade to Rails 6

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -100,6 +106,25 @@
...
+  # Use a different logger for distributed setups.
+  # require "syslog/logger"
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
+
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
+    logger           = ActiveSupport::Logger.new(STDOUT)
+    logger.formatter = config.log_formatter
+    config.logger    = ActiveSupport::TaggedLogging.new(logger)
+  end
+
```
Comment (pixeltrix) [inline config/environments/production.rb:127]: Can you remove the extra log config that's been added here - our custom config is further up the file

File: config/environments/test.rb [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
-  # Configure static file server for tests with Cache-Control for performance.
+  # Configure public file server for tests with Cache-Control for performance.
   config.public_file_server.enabled = true
   config.public_file_server.headers = {
-    "Cache-Control" => "public, max-age=#{1.hour.to_i}"
+    'Cache-Control' => "public, max-age=#{1.hour.to_i}"
```
Comment (pixeltrix) [inline config/environments/test.rb:21]: Can you remove the `#{1.hour.to_i}` and replace with a static `3600`. Once you've done that remove the require at the top of the file. DHH likes to have his duration things but it's pointless really.

File: config/initializers/date_format.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Time::DATE_FORMATS[:stepped_cache_key] = lambda { |time|
+  time
+    .change(sec: (time.sec.div(5) * 5))
+    .utc.to_s(:usec)
+}
```
Comment (pixeltrix) [inline config/initializers/date_format.rb:5]: Worked out a better way of doing this:

```
Time::DATE_FORMATS[:stepped_cache_key] = lambda { |time| (time.floor - time.to_i % 5).to_s(:usec) }
```

Forgot that Rails 6.1 backports `Time#floor` to earlier rubies.

File: lib/arel_extensions.rb [hunk 4]
```diff
@@ -41,11 +41,11 @@ def visit_Arel_Nodes_Contained(o, collector)
         infix_value(o, collector, " <@ ")
       end
 
-      def visit_Arel_Nodes_Contains(o, collector)
+      def visit_Arel_Nodes_Containz(o, collector)
         infix_value(o, collector, " @> ")
       end
 
-      def visit_Arel_Nodes_Overlaps(o, collector)
+      def visit_Arel_Nodes_Overlapz(o, collector)
```
Comment (pixeltrix) [inline lib/arel_extensions.rb:48]: This whole module can go because rails/rails@3a8eee9

File: lib/tasks/errors.rake [hunk 5]
```diff
@@ -9,7 +9,7 @@ namespace :errors do
       end
     end
 
-    context_class = Class.new(ActionView::Base) do
+    context_class = Class.new(ActionView::Base.with_empty_template_cache) do
```
Comment (pixeltrix) [inline lib/tasks/errors.rake:12]: Nice 👍

File: spec/requests/csrf_token_spec.rb [hunk 6]
```diff
@@ -19,7 +19,9 @@
     one_time_pad.bytes.zip(real_csrf_token.bytes).map { |(c1,c2)| c1 ^ c2 }.pack('c*')
   end
 
-  let(:real_csrf_token) { Base64.strict_decode64(session[:_csrf_token]) }
+  let(:real_csrf_token) do
+    Base64.urlsafe_decode64(session[:_csrf_token])
+  end
```
Comment (pixeltrix) [inline spec/requests/csrf_token_spec.rb:24]: Can we use `{ }` here - it's only 1 character longer 😄

**Review by pixeltrix — CHANGES_REQUESTED** [review]

File: config/environments/production.rb [hunk 7]
```diff
@@ -1,3 +1,5 @@
+require "active_support/core_ext/integer/time"
```
Comment (freesteph) [inline config/environments/production.rb:1]: whoops

Comment (freesteph) [reply config/environments/production.rb:1]: I'm not sure why that was added; can be removed right?

**Review by pixeltrix — APPROVED** [review]

File: config/environments/production.rb (hunk 7 above)
Comment (pixeltrix) [reply config/environments/production.rb]: yes

Comment (pixeltrix) [reply config/environments/production.rb]: As long as `1.hour.to_i` is no where in the file

**Pull request merged**